pkg/p2p/libp2p/protocols_test.go

```diff
@@ -141,7 +141,34 @@ func TestNewStream_semanticVersioning(t *testing.T) {
 			expectErrNotSupported(t, err)
 		}
 	}
+}
+
+func TestDisconnectError(t *testing.T) {
```

Comment:
Hmm, I thought this was before my second commit?

Reply:
I will try to run int a couple of more times, but I assume this is because the disconnect happened before the headers were exchanged as well, which is not relevant for this test. So, I added a commit ignoring the actual error from stream. Maybe I can check for the specific `send headers` error, that would be a bit more accurate

Reply:
Yes, could you add a comment with the reason why the error is not checked? Just to make it clear that it is because of the disconnect.

Reply:
Sounds good. I will add the comment then.

Reply:
Perfect, thanks.